routing.go

```diff
@@ -539,14 +539,22 @@ func (dht *IpfsDHT) FindProviders(ctx context.Context, c cid.Cid) ([]peer.AddrIn
 
 // FindProvidersAsync is the same thing as FindProviders, but returns a channel.
 // Peers will be returned on the channel as soon as they are found, even before
-// the search query completes.
+// the search query completes. If count is zero then the query will run until it
+// completes. Note: not reading from the returned channel may block the query
+// from progressing.
 func (dht *IpfsDHT) FindProvidersAsync(ctx context.Context, key cid.Cid, count int) <-chan peer.AddrInfo {
-	peerOut := make(chan peer.AddrInfo, count)
 	if !dht.enableProviders {
+		peerOut := make(chan peer.AddrInfo)
 		close(peerOut)
 		return peerOut
 	}
 
+	chSize := count
+	if count == 0 {
+		chSize = 16
```

Comment:
factor out to a constant?

Reply:
or let's just make this 1 until we find a reason to increase it.

Reply:
1 should be fine.

Reply:
1 should be fine.

Reply:
done